mint-client/src/clients/user.rs

```diff
@@ -200,26 +203,30 @@ impl UserClient {
         })
     }
 
-    pub async fn peg_out<R: RngCore + CryptoRng>(
+    // FIXME: is this comment still true?
+    /// **ATTENTION**: calling this function multiple times without committing the batch to the
+    /// database is not supported and will result in an accidental double spend.
+    async fn create_coin_input<R: RngCore + CryptoRng>(
         &self,
-        amt: bitcoin::Amount,
-        address: bitcoin::Address,
+        amount: Amount,
+        rng: &mut R,
+    ) -> Result<(Vec<KeyPair>, Coins<Coin>), ClientError> {
+        let coins = self.select_and_spend_coins(amount, rng).await?;
+        let (spend_keys, coins) = self.mint_client().create_coin_input_from_coins(coins)?;
+        Ok((spend_keys, coins))
+    }
+
+    async fn sign_and_submit_tx<R: RngCore + CryptoRng>(
```

Comment:
Transaction construction, signing and submission was now repeated 3 times in this struct so I made a method for it.